### DISCUSSION TITLE
using hvcc with juce

### DISCUSSION THREAD

**Discussion by shakfu** [Wasted-Audio/hvcc#160] [Show and tell]
FYI, I just saw this project: https://github.com/o-g-sus/JUCE-HEAVY

> Demo / Template JUCE Project showing how to connect JUCE with Heavy C++ (HVCC) generated Source files

**Comment by dromer** [comment]
Correct. It was presented on the Heavy discord already.

I've converted this to a discussion as it is not an issue with hvcc.

**Comment by dromer** [comment]
Someone also started an effort to create a c2juce generator: https://github.com/cameronbroe/hvcc/tree/add-juce-generator

**Reply by shakfu** [reply]
Thanks, they are quite different: the first is a really a demo with specialized code and the latter is more generic and fits more with hvcc. In any case, both look promising!

**Reply by dromer** [reply]
They are not _that_ different. The first uses the plain c/c++ output, the second creates an actual integration with the compiler so is much more useful in the long run.

I would consider future integration of the second one, but I personally won't support it since JUCE is quite complicated and I don't want to deal with it. (it's also not liberally licensed which should be clearly documented)

**Reply by shakfu** [reply]
Well in any case, the timing is good since I just bought a [book](https://www.theaudioprogrammer.com/synth-plugin-book) on Juce plugin development, so I'll be interested either way.
